controllers/ibmpowervscluster_controller_test.go

```diff
@@ -839,6 +871,399 @@ func TestIBMPowerVSClusterReconciler_delete(t *testing.T) {
 			g.Expect(clusterScope.Client.Update(ctx, powervsImage2)).To(Not(Succeed()))
 		})
 	})
+
+	t.Run("when delete TransitGateway returns error", func(t *testing.T) {
+		g := NewWithT(t)
+		clusterScope = powervsClusterScope()
+		clusterScope.IBMPowerVSCluster.Status.TransitGateway = &infrav1beta2.TransitGatewayStatus{
+			ID:                ptr.To("transitgatewayID"),
+			ControllerCreated: ptr.To(true),
+			PowerVSConnection: &infrav1beta2.ResourceReference{
+				ControllerCreated: ptr.To(false),
+				ID:                ptr.To("connectionID"),
+			},
+			VPCConnection: &infrav1beta2.ResourceReference{
+				ControllerCreated: ptr.To(false),
```

Comment:
```suggestion
				ControllerCreated: ptr.To(true),
```
Ideally, when TG is created by controller, these would be true as well.
Please change for below case as well.

Reply:
Done